### PR TITLE
ref(snql) Add test to ensure splitter is running correctly

### PR DESCRIPTION
**Pull request by evanh**
One of the discover queries had a SQL mismatch because the dry run version of SnQL wasn't using the splitter. There's nothing broken with SnQL and the splitter but I added a test just to be sure.

### CODE REVIEW

File: tests/test_discover_api.py [hunk 1]
```diff
@@ -1604,6 +1604,53 @@ def test_exception_stack_column_boolean_condition(self) -> None:
...
+                    "aggregations": [],
+                    "consistent": False,
+                    "debug": False,
+                }
+            ),
+            entity="discover",
+        )
+        assert response.status_code == 200
+        data = json.loads(response.data)
+        assert "PREWHERE in(event_id, tuple" in data["sql"]
```
Comment (lynnagara) [inline tests/test_discover_api.py:1652]: Can we assert on the entire PREWHERE expression?

Comment (evanh) [reply tests/test_discover_api.py:1652]: If you'd rather, I can do that.

Comment (fpacifici) [reply tests/test_discover_api.py:1652]: I don't think this is a great approach to do this test.
First: I am not sure, by reading the test name `test_extra_group_conditions`, what this is supposed to test. We could also make a case that splitter is an implementation detail that should not be tested by an integration test that only checks that the output matches what is expected for an input.
Second we are depending on the SQL parameter in the response which we know is broken. It does not reflect the splitter (it includes only one of the queries).

If we are looking for a specific issue with the splitter, can't we build a splitter unit test ?

**Comment by evanh** [comment]
Turns out this isn't the right way to test this.

**Pull request closed without merging**